tests/lru_cache_test.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+import string
+import tempfile
+
+from absl.testing import absltest
+
+from jax._src.lru_cache import LRUCacheImpl
+import jax._src.test_util as jtu
+
+
+class LRUCacheTest(jtu.JaxTestCase):
+
+    def test_cache_eviction(self):
+        def generate_random_k():
+            # simulate keys with many collisions
+            return random.choice(string.ascii_lowercase[:12])
+
+        def generate_random_v(k):
+            # simulate large values while ensures that one k corresponds to one v
+            return bytes(f"{k}abcdefghijklmnopqrstuvwxyz" * 8192, encoding="utf-8")
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            cache = LRUCacheImpl(tmpdirname, 884700)
+
+            for _ in range(100000):
+                k = generate_random_k()
+
+                if cache.get(k) is not None:
+                    pass  # cache hit
+                else:
+                    v = generate_random_v(k)
+                    cache.put(k, v)
```

Comment:
Worth checking that `tmpdir` does not have more than `max_size` files.

Reply:
Done